morango/syncsession.py

```diff
@@ -134,35 +152,78 @@ def certificate_signing_request(self, parent_cert, scope_definition_id, scope_pa
         csr_cert.save()
         return csr_cert
 
-        pass
-
     def _get_certificate_chain(self, server_cert):
         # get ancestors certificate chain for this server cert
         cert_chain_resp = self._request(api_urls.CERTIFICATE, params={'ancestors_of': server_cert.id})
 
         # upon receiving cert chain from server, we attempt to save the chain into our records
         Certificate.save_certificate_chain(cert_chain_resp.data, expected_last_id=server_cert.id)
 
-    def _create_transfer_session(self, push, filter_prefixes):
+    def _create_transfer_session(self, data):
+        # create transfer session on server
+        return self._request(api_urls.TRANSFERSESSION, method="POST", data=data)
+
+    def _update_transfer_session(self, data, transfer_session):
+        # update transfer session on server side with kwargs
+        return self._request(api_urls.TRANSFERSESSION, method="PATCH", lookup=transfer_session.id, data=data)
+
+    def _close_transfer_session(self, transfer_session):
+        # "delete" transfer session on server side
+        return self._request(api_urls.TRANSFERSESSION, method="DELETE", lookup=transfer_session.id)
+
+    def _close_sync_session(self, sync_session):
+        # "delete" sync session on server side
+        return self._request(api_urls.SYNCSESSION, method="DELETE", lookup=sync_session.id)
+
+
+class SyncClient(object):
+    """
+    Controller to support client in initiating syncing and performing related operations.
+    """
+    def __init__(self, sync_connection, sync_session, profile=''):
+        self.sync_connection = sync_connection
+        self.sync_session = sync_session
+        self.current_transfer_session = None
+        self.profile = profile
+
+    def close_sync_session(self):
+
+        # "delete" sync session on server side
+        self.sync_connection._close_sync_session(self.sync_session)
+
+        # "delete" our own local sync session
+        self.sync_connection.sync_session.active = False
+        self.sync_connection.sync_session.save()
+        self.sync_connection.sync_session = None
+        self = None
```

Comment:
As it's just a local variable, I don't think this does anything?

Reply:
can update.